### DISCUSSION TITLE
复式记账的自动化探索实践 | 高策

### DISCUSSION THREAD

**Discussion by giscus** [gaocegege/blog-comments#15] [General]
# 复式记账的自动化探索实践 | 高策

江湖小虾米

http://gaocegege.com/Blog/%E9%9A%8F%E7%AC%94/double-entry

**Comment by Endle** [comment]
我有一个问题

```
  --config ./example/wechat/config.yaml \
  --provider wechat \
```

这个 config 和 provider 有什么关系？比如我想给某个银行写一个新的 provider，我初步的理解是，我只要照猫画虎写一个 yaml 文件，接着让 double-entry-generator 去读即可，也不需要重新编译代码。

我看了几个示例，有一个猜测：同一个配置文件里可以塞多个provider，所以要选择，读取哪一段的配置。比如我想用这个文件，我就要设置provider huobi
https://github.com/deb-sig/double-entry-generator/blob/864a4f6e20da637bc966469a1824f654591c9a2a/example/huobi/config.yaml#L7

**Reply by gaocegege** [reply]
@Triple-Z

cc 懂哥

**Reply by Endle** [reply]
刚才我写的博客引用了一下这个工具 https://blog.zhenbo.pro/2023/01/21/write-beancount-csv-importer-from-scratch/

**Reply by Triple-Z** [reply]
同一个 config 确实可以同时服务多种 provider（只要你每种 provider 需要使用的全局配置 `Default*` 都一样即可），每种 provider 自身的配置都在其自己名下的对象中（如微信支付配置会在`wechat` 对象下）。

如果需要新增 provider，根据目前 deg 的设计，还是需要编写 go 代码并重新编译的。如 README 中的这个架构图所示：当你需要新增 provider 时，需要在代码中同时新增 `provider` 和 `analyzer`。

```
┌───────────┐  ┌──────────┐  ┌────┐  ┌──────────┐  ┌──────────┐
│ translate │->│ provider │->│ IR │->│ compiler │->│ analyser │
└───────────┘  └──────────┘  └────┘  └──────────┘  └──────────┘
                  alipay               beancount      alipay
                  wechat                              wechat
                  huobi                               huobi
```

**Reply by Endle** [reply]
谢谢解答！
